docs/guides/ci-cd.md

```diff
@@ -1,103 +1,218 @@
 [← Back to Home](../index.md)
 
-# CI/CD Integration
+# CI/CD Guide for llm-application Test Library
 
-llm_app_test is designed for seamless integration with CI/CD pipelines.
+This guide provides a detailed explanation of setting up Continuous Integration (CI) using GitHub Actions for the `llm-application` test library. It includes a step-by-step example of a CI workflow, environment variable setup, and best practices for CI integration.
 
-## Setting Up CI/CD
+---
 
-### 1. Environment Setup
-Add your API keys as secrets in your CI/CD environment:
+## 1. CI Integration
 
-```
-env:
-  OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+### GitHub Actions Example
+Below is an example of a GitHub Actions workflow file (`ci.yml`) configured for the `llm-application` test library. It runs necessary tests in parallel for changes pushed to the `main` branch or submitted as pull requests to `main`.
 
-# OR
-env:
-  ANTHROPIC_API_KEY: ${{ secrets.ANTHROPIC_API_KEY }}
-```
+```yaml
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
 
-### 2. Test Configuration
+jobs:
+  test:
+    name: Run basic tests and ensure all pass
+    runs-on: ubuntu-latest
 
-```
-from llm_app_test.behavioral_assert.behavioral_assert import BehavioralAssertion
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
 
-def test_llm_output():
-    behavioral_assert = BehavioralAssertion() # Your tests here
-```
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install GPG
+        run: sudo apt-get install -y gpg
+
+      - name: Decrypt API key
+        env:
+          GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
+        run: |
+          gpg --batch --yes --passphrase "$GPG_PASSPHRASE" \
+              -o openai-api-key.txt -d openai-api-key.txt.gpg
 
+      - name: Set API Key Environment Variable
+        run: |
+            export OPENAI_API_KEY=$(cat openai-api-key.txt)
+            echo "OPENAI_API_KEY=$OPENAI_API_KEY" >> $GITHUB_ENV
 
-### 3. CI/CD Pipeline Configuration
+      - name: Install dependencies
+        run: |
+          pip install .
```

Comment:
`pip install .` will only work for us. Users have to use `pip install llm-app-test`

Reply:
done.